src/scripts/checkMarkdownLanguage.mjs

```diff
@@ -0,0 +1,91 @@
+import { spawn } from "child_process"
+
+const commonDiffCommand = [
+  "diff",
+  "--cached",
+  "--name-only",
+  "--diff-filter",
+  "AM",
+]
+
+const getMarkdownFilesWithLanguage = () => {
+  return new Promise((resolve, reject) => {
+    const markdownFilesWithLanguageCommand = spawn("git", [
+      ...commonDiffCommand,
+      "-G",
+      "lang:",
+      "--",
+      "*.md",
+    ])
+
+    markdownFilesWithLanguageCommand.stdout.on("data", (data) => {
+      console.log(data.toString("utf8"))
+      resolve(data)
+    })
+
+    markdownFilesWithLanguageCommand.stderr.on("data", (data) => {
+      reject(data)
+    })
+
+    markdownFilesWithLanguageCommand.on("error", (error) => {
+      reject(error)
+    })
+  })
+}
+
+const getAllMarkdownFiles = () => {
+  return new Promise((resolve, reject) => {
+    const allMarkdownFilesCommand = spawn("git", [
+      ...commonDiffCommand,
+      "--",
+      "*.md",
+    ])
+
+    allMarkdownFilesCommand.stdout.on("data", (data) => {
+      console.log(data.toString("utf8"))
+      resolve(data)
+    })
+
+    allMarkdownFilesCommand.stderr.on("data", (data) => {
+      reject(data)
+    })
+
+    allMarkdownFilesCommand.on("error", (error) => {
+      reject(error)
+    })
+  })
+}
+
+const doAllMarkdownsHaveALanguage = async () => {
+  try {
+    const allMarkdownFilesBuffer = await getAllMarkdownFiles()
+    const markdownFilesWithLanguageBuffer = await getMarkdownFilesWithLanguage()
+    const comparisonResult = Buffer.compare(
```

Comment:
The file names are printed to the STDOUT and the same is reported by NodeJS as a Buffer.
If the file names are the same then the Buffer should be the same too.

Reply:
(Assuming that while this process is running, no other process printed something onto the STDOUT [highly unlikely])